### ISSUE TITLE
[docdb] Disable retries for the master flush and compaction async tasks

### ISSUE THREAD

**Issue by bmatican** [yugabyte/yugabyte-db#4231]
These are long running tasks, that could independently timeout. Moreover, retrying these can actually be counter productive.

Note, this is related to, but different from #1847, which is specifically about yb-admin not waiting on if these background tasks were aborted or timed out.

**Comment by amitanandaiyer** [comment]
RetryingTSRpcTask::num_max_retries can be appropriately overridden

**Comment by amitanandaiyer** [comment]
Also, we seem to already have
```
 // Returns true if we should not retry for this task type.
  bool NoRetryTaskType() {
    return type() == ASYNC_FLUSH_TABLETS;
  }
```

**Comment by bmatican** [comment]
Oh good point, seems like we fixed this in #2229, thanks for pointing this out @amitanandaiyer
